test/BotTest.js

```diff
@@ -618,4 +618,52 @@ describe('Bot', function () {
             expect(classifier).toBe(fakeClassifier);
         });
     });
-});
\ No newline at end of file
+
+    describe('ChainableMethods', function () {
+        it('Chainable methods with train and addSkill', function (done) {
+            var mockClassifier = mockClassifierWithMockClassifierFactory();
```

Comment:
You shouldn't need the mockClassifier here if you are not going to use it. The classifier is only used when you call the resolve method which, from what I can see is not called in this test.